server/save.py

```diff
@@ -309,6 +308,13 @@ def upgrade(version):
             )
         db.foreign_keys = True
         Constants.get().update(save_version=Constants.save_version + 1).execute()
+    elif version == 21:
+        db.foreign_keys = False
+        db.execute_sql(
+            'CREATE TABLE IF NOT EXIST "marker" ("id" INTEGER NOT NULL PRIMARY KEY, `uuid` TEXT NOT NULL , `user_id` INTEGER NOT NULL, `location_id` INTEGER NOT NULL, FOREIGN KEY(`uuid`) REFERENCES `shape`(`uuid`) ON DELETE CASCADE, FOREIGN KEY ("location_id") REFERENCES "location" ("id") ON DELETE CASCADE, FOREIGN KEY(`user_id`) REFERENCES `user`(`id`) ON DELETE CASCADE)'
```

Comment:
```suggestion
            'CREATE TABLE IF NOT EXISTS "marker" ("id" INTEGER NOT NULL PRIMARY KEY, `uuid_id` TEXT NOT NULL, `user_id` INTEGER NOT NULL, `location_id` INTEGER NOT NULL, FOREIGN KEY(`uuid_id`) REFERENCES `shape`(`uuid`) ON DELETE CASCADE, FOREIGN KEY ("location_id") REFERENCES "location" ("id") ON DELETE CASCADE, FOREIGN KEY(`user_id`) REFERENCES `user`(`id`) ON DELETE CASCADE)'
```

Reply:
There was a typo in the exists keyword and the uuid field was missing its _id postfix

Reply:
Changed